### DISCUSSION TITLE
How to turn `GraphicsPath` into `SvgPath` or some sort of drawable on canvas?

### DISCUSSION THREAD

**Discussion by ryuujo1573** [vvvv/SVG#908] [Q&A]
I was dizzy about implementing my jigsaw puzzle generator.

I'll draw some curves as jigsaw sides in Ai and export as `.svg`, and I have no idea of harvesting my curves in drawing jigsaw puzzles automatically.

Thank you for your work, and uhhh, if it is convenient for you, any reply would help!

**Comment by ryuujo1573** [comment]
```C#
public void SvgTest()
{
    var path = new GraphicsPath();
    path.AddBezier(
        0f, 0f,
        0f, 4.8f,
        0f, 4.8f,
        4.8f, 4.8f
        );
    var svgPath = new SvgPath();
    var segment = new SvgMoveToSegment(new PointF(0, 0));
    segment.AddToPath(path);
    svgPath.PathData = new SvgPathSegmentList();
    svgPath.PathData.Add(segment);

    var hBitmap = new Bitmap(1000, 1000).GetHbitmap(Color.WhiteSmoke);
    var img = Image.FromHbitmap(hBitmap);
    svgPath.RenderElement(SvgRenderer.FromImage(img));
    img.Save("test.bmp");
}
```

Why this doesn't work and outputs an empty bmp/png?

**Comment by H1Gdev** [comment]
I don't think it's appropriate to generate from `GraphicsPath`.
What kind of SVG do you want to generate ?

For example, why not try using `SvgPathBuilder`?

**Reply by ryuujo1573** [reply]
Okay, I will try as you hint asap
btw without easy-read documentation I found myself struggling with IDE.. Is there any way to learn this framework?
Thank you for your early reply.

**Reply by mrbean-bremen** [reply]
Documentation is indeed a weak point of this library. Anyone is welcome to contribute to the documentation, of course, but currently the project lacks somebody who is able and has time to create such documentation.

**Reply by H1Gdev** [reply]
At the moment, I think fastest way is to refer to [sample codes](http://svg-net.github.io/SVG/doc/Samples.html).

**Reply by ryuujo1573** [reply]
THANK YOU!!!
Well I am a newbee of dotNET lover, and if I can be fluent in this proj, I guess I would be willing to be one of documentors...
Gracious!
